### ISSUE TITLE
Key Error Issue when running pgsync

### ISSUE THREAD

**Issue by CE-Creates** [toluaina/pgsync#59]
Traceback (most recent call last):
  File "/usr/local/bin/pgsync", line 7, in <module>
    sync.main()
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pgsync/sync.py", line 1012, in main
    sync.pull()
  File "/usr/local/lib/python3.7/site-packages/pgsync/sync.py", line 883, in pull
    self.logical_slot_changes(txmin=txmin, txmax=txmax)
  File "/usr/local/lib/python3.7/site-packages/pgsync/sync.py", line 241, in logical_slot_changes
    self.sync_payloads(payloads)
  File "/usr/local/lib/python3.7/site-packages/pgsync/sync.py", line 743, in sync_payloads
    payloads,
  File "/usr/local/lib/python3.7/site-packages/pgsync/sync.py", line 390, in _payloads
    value = payload_data[key]
KeyError: 'projectHeroMediumId'

**Comment by toluaina** [comment]
Would you mind providing details of your database schema and pgsync config?

**Comment by somechild** [comment]
Amazing work here @toluaina !
Thought I'd set this up and try it out today, but ran into a similar error:
```
Polling db lroll: 0 cache items
Exception in thread Thread-16:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pgsync/sync.py", line 782, in poll_redis
    self.on_publish(payloads)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pgsync/sync.py", line 849, in on_publish
    self.sync_payloads(_payload)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pgsync/sync.py", line 742, in sync_payloads
    for doc in self._payloads(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pgsync/sync.py", line 392, in _payloads
    value = payload_data[key]
KeyError: 'storyId'
Polling db lroll: 2 cache items
Polling db lroll: 2 cache items
Polling db lroll: 2 cache items
```

`storyId` is the pk of my postgres table

pgsync config is as follows:
```
[
    {
        "database": "lroll",
        "index": "story",
        "setting": {
            "analysis": {
                "analyzer": {
                    "autocomplete": {
                        "filter": [
                            "lowercase"
                        ],
                        "tokenizer": "autocomplete"
                    }
                },
                "tokenizer": {
                    "autocomplete": {
                        "type": "edge_ngram",
                        "min_gram": 3,
                        "max_gram": 10,
                        "token_chars": [
                            "letter",
                            "digit"
                        ]
                    }
                }
            }
        },
        "nodes": [
            {
                "table": "story",
                "schema": "public",
                "columns": [
                    "storyId",
                    "storyTitle",
                    "dateCreated",
                    ...
                ],
                "children": [
                    {
                        "table": "clip",
                        "columns": [
                            "clipId",
                            "creatorId",
                            ....
                        ],
                        "label": "clips",
                        "relationship": {
                            "variant": "object",
                            "type": "one_to_many"
                        }
                    }
                ]
            }
        ]
    }
]
```

**Comment by toluaina** [comment]
@somechild Would you mind sending me your database config, please? I already have your pgsync schema

**Comment by CE-Creates** [comment]
We are still seeing these errors quite often. Since we havent moved our app to production, we have the luxury to delete the index and reindex it when this seems to have gone. But we are concerned about how this issue would be addressed for a production app. 

@toluaina Could you please print more details when the program encounter this error? Is this error due to a particular record? If so, could you please print the data and skip and continue handling rest of the changes?

**Comment by somechild** [comment]
@toluaina By database config are you looking for the `.env`?

```
SCHEMA='public'
QUERY_CHUNK_SIZE=1000
POLL_TIMEOUT=10
REPLICATION_SLOT_CLEANUP_INTERVAL=3600.0
ELASTICSEARCH_SCHEME='https'
ELASTICSEARCH_HOST='a****net'
ELASTICSEARCH_PORT=443
ELASTICSEARCH_USER='*****'
ELASTICSEARCH_PASSWORD='****'
ELASTICSEARCH_TIMEOUT=10
ELASTICSEARCH_CHUNK_SIZE=2000
ELASTICSEARCH_MAX_CHUNK_BYTES=104857600
ELASTICSEARCH_THREAD_COUNT=4
ELASTICSEARCH_QUEUE_SIZE=4
PG_HOST='***-1.rds.amazonaws.com'
PG_USER='****'
PG_PORT=5432
PG_PASSWORD='*****'
PG_SSLMODE='allow'
REDIS_HOST='red***s.com'
REDIS_PORT=14652
REDIS_DB='re***907'
REDIS_AUTH='****'
REDIS_CHUNK_SIZE=100
REDIS_SOCKET_TIMEOUT=10
```

**Comment by toluaina** [comment]
@somechild Sorry I meant your database schema or just dump the db without data. This could be related to the issue @CE-Creates  is reporting.

**Comment by toluaina** [comment]
Sorry about the delay. The KeyEror has been resolved in v 1.1.24. You would need to re-run the bootstrap using the schema config.  i.e `bootstrap -c /path.to/config`

**Comment by CE-Creates** [comment]
This is solved in 1.1.24 - Verified
